extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java

```diff
@@ -126,14 +139,22 @@ public void onLightStateChanged(HueBridge bridge, FullLight light) {
 
     private ThingUID getThingUID(FullLight light) {
         ThingUID bridgeUID = hueBridgeHandler.getThing().getUID();
-        ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, light.getModelID().replaceAll("[^a-zA-Z0-9_]", "_"));
+        ThingTypeUID thingTypeUID = getThingTypeUID(light);
 
-        if (getSupportedThingTypes().contains(thingTypeUID)) {
+        if (thingTypeUID != null && getSupportedThingTypes().contains(thingTypeUID)) {
             String thingLightId = light.getId();
-            ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID, thingLightId);
+            ThingTypeUID modelUID = new ThingTypeUID(BINDING_ID,
```

Comment:
isn't this line outdated? Why do we need to construct another ThingTypeUID here?